tools/build/Makefile.vars

```diff
@@ -286,7 +286,9 @@ COMMON_CFLAGS += -g -fno-lto
 endif
 
 ifeq (y,$(BUILD_TYPE_RELEASE))
-COMMON_CFLAGS += $(RELEASE_COMMON_CFLAGS)
+ifeq (,$(shell echo $(COMMON_CFLAGS) | grep -e "-O0"))
+COMMON_CFLAGS += $(FORTIFY_CFLAGS)
+endif
```

Comment:
OK, so I understand why I can't use ```BUILD_TYPE_RELEASE=n make```. It's because Makefile include .config, and in .config BUILD_TYPE_RELEASE is set explicitly to ```y```, not via ```?=```, so the environment variable is being ignored. Thus, my only option, if I wish not to modify the .config file is to use COMMON_CFLAGS which, in turn, leaves me with the solution in this PR.